docs/rfc/0002-webrc-client-api.md

```diff
@@ -0,0 +1,1612 @@
+```
+                                                                       .------------Server-----------.
+                                                                       :     .-------------------.   :
+                          .--------------------------------------------+-----o  Control Service  :   :
+                          :                                            :     '--------o----------'   :
+                          :                                            :              |              :
+                          :                                            :        Control Api          :
+.--------Client-----------+------------------------.                   :              |              :
+:  .--------------------. :  .--------------------. :  .-Client-API--. :  .-----------o------------. :
+:  :  User Application  o-'  :     Web Client     o-+--'             '-+--o      Media Server      : :
+:  :                    :----:                    o-+--.             .-+--o                        : :
+:  '--------------------'    '--------------------' :  '----Media----' :  '------------------------' :
+'---------------------------------------------------'                  '-----------------------------'
+```
+
+So, how it works from `Media Server` point of view:
+1. `Control Service` configures media room via `Control API`.  
+2. `Media Server` provides all necessary information (URLs + credentials) for all room members.
+3. `User Application` passes credentials and other necessary stuff (like `<video>` elements) to `Web Client`.
+4. And voilà!
+
+
+### Transport considerations
+
+Although, signalling can be implemented on top of any transport, WebSocket suits the most since it provides small overhead reliable duplex connection, is widely adopted and supported.
+
+
+### Protocol considerations
+
+Existing best practices are recommended for final implementation:
+1. Message level `ping`/`pong`, since it is the most reliable way to detect connection loss (protocol level WebSocket `ping`/`pong` may disfunct due to browser implementation and is not exposed to `Web Client`).
+2. Reconnects, since [RTCPeerConnection] always outlives WebSocket connection in cases of network issues, and both parts should know when to dispose related resources.
+3. Transactions, so each part will receive answers to all request sent, and be able to know if request was processed correctly or caused some kind of error.
+4. Using custom Close Frame Status Codes, to implement reliable send-and-close.
+
+Transactions:
+
+Each message is represented as:
+
+```rust
+struct WsMessage<T> {
+    id: i64,
+    payload: Option<Result<Payload<T>, Error>>,
+}
+
+struct Payload<T> {
+    method: String,
+    params: T,
+}
+```
+
+Each message requires answer. Answer can carry some payload (e.g. answering with [SDP Answer] to [SDP Offer]), error, or just noting, which just means that message reached destination and was processed.
```

Comment:
@alexlapa this requires more formalization. Open questions:

1. Response message is not specified. Request message has `id`. How can I distinguish which response to which request belongs?

2. Ordering is not specified. Is it sync (req1 -> resp1 -> req2 -> resp2), or async (req1 -> req2 -> resp2 -> resp1)?